### ISSUE TITLE
explicit_iter_loop warning message not helpful enough to fix the issue

### ISSUE THREAD

**Issue by gnzlbg** [Manishearth/rust-clippy#1493]
In the following code: 

```rust
fn u8_runner<F: Fn(u8) -> u8>(bench: &mut Bencher, f: F) {
    let mut vs = (0..u8::max_value()).collect::<Vec<_>>();
    bench.iter(|| {
        for mut v in vs.iter_mut() {
            *v = bencher::black_box(f(bencher::black_box(*v)));
        }
    })
}
```

I get the following lint: 

> `warning: it is more idiomatic to loop over `&mut vs` instead of `vs.iter_mut()`, #[warn(explicit_iter_loop)] on by default`

From this it is not clear at all what clippy wants for me. 

I've tried 

```rust
fn u8_runner<F: Fn(u8) -> u8>(bench: &mut Bencher, f: F) {
    let mut vs = (0..u8::max_value()).collect::<Vec<_>>();
    bench.iter(|| {
        for &mut v in vs {
            *v = bencher::black_box(f(bencher::black_box(*v)));
        }
    })
}
```

and 

```rust
fn u8_runner<F: Fn(u8) -> u8>(bench: &mut Bencher, f: F) {
    let mut vs = (0..u8::max_value()).collect::<Vec<_>>();
    bench.iter(|| {
        for &mut v in vs.iter() {
            *v = bencher::black_box(f(bencher::black_box(*v)));
        }
    })
}
```

without any luck. Reading [the online description](ttps://github.com/Manishearth/rust-clippy/wiki#explicit_iter_loop) was not helpful. Maybe clippy should really tell me what the X, Y, and Z in the following snippet should look like:

```rust
fn u8_runner<F: Fn(u8) -> u8>(bench: &mut Bencher, f: F) {
    let mut vs = (0..u8::max_value()).collect::<Vec<_>>();
    bench.iter(|| {
        for X?? in Y?? {
            Z?? = bencher::black_box(f(bencher::black_box(Z??)));
        }
    })
}
```

where:

- X is the iteration variable,
- Y is what i am iterating on,
- Z is how I am accessing the value.

**Comment by oli-obk** [comment]
latest clippy already has some improvements in that direction, and https://github.com/Manishearth/rust-clippy/pull/1491 should fix the last of them.

**Comment by gnzlbg** [comment]
@oli-obk is it released already? I just checked and I am trying with the latest release.

**Comment by oli-obk** [comment]
nope, the PR isn't even merged yet ;)

**Comment by oli-obk** [comment]
note that clippy tried to tell you to use `for v in &mut vs`
